go/stats/prometheusbackend/prometheusbackend.go

```diff
@@ -67,8 +64,11 @@ func (be PromBackend) publishPrometheusMetric(name string, v expvar.Var) {
 		newMultiTimingsCollector(st, be.buildPromName(name))
 	case *stats.Histogram:
 		newHistogramCollector(st, be.buildPromName(name))
+	case *stats.String, stats.StringFunc, stats.StringMapFunc, *stats.Rates:
```

Comment:
Missing a few:

JSONFunc, Float, FloatFunc, StringMap

FWIW, I just looked up everything that implemented String() inside the stats package, which is what's required for the expvar interface.

Reply:
Oh good catch! I only checked for types that were actually used by the vitess code today.

@michael-berlin Since these types aren't used anywhere my instinct is that we should remove them from `stats` rather than either add them to this list or run the risk that someone starts using them in the future and then it will panic for anyone using prometheus.